cmd/plugin/vault-auth-spire.go

```diff
@@ -133,11 +142,12 @@ func parseSettings() (*common.Settings, error) {
 	settingsFlags.StringVar(&settingsFilePath, "settings-file", "", "Path to plugin settings")
 	settingsFlags.Parse(os.Args[1:])
 
-	if settings, err := common.ReadSettings(settingsFilePath); err != nil {
+	settings, err := common.ReadSettings(settingsFilePath)
+	if err != nil {
 		return nil, errors.New("vault-auth-spire: Failed to read settings from '" + settingsFilePath + "' - " + err.Error())
```

Comment:
Seems odd to return this error here, why not have `common.ReadSettings()` return useful errors, and then just `return common.ReadSettings()`?

Reply:
I think it makes more sense to propagate errors in `common.ReadSettings` and then wrap them in this method so you don't have to rewrite "failed to read from path xxx" for each error in `common.ReadSettings`.

Reply:
Fair enough